print/libpaper/Portfile

```diff
@@ -1,55 +1,35 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem      1.0
+PortGroup       github 1.0
+
+github.setup    rrthomas libpaper 2.1.2 v
+revision        0
+checksums       rmd160  6b3898d28155228927fa78f55bc747fbcd8e0bc2 \
+                sha256  1fda0cf64efa46b9684a4ccc17df4386c4cc83254805419222c064bf62ea001f \
+                size    1264200
+
+github.tarball_from releases
 
-name            libpaper
-set mainversion 1.1.29
-# Keep in case need again in the future
-#set nmuversion  5
-#set realversion ${mainversion}+nmu${nmuversion}
-#version         ${mainversion}.${nmuversion}
-set realversion ${mainversion}
-version         ${mainversion}
 categories      print
-license         GPL-2
+# libpaper is LGPLv2.1+, paperspecs is public domain
+license         LGPL-2.1 public-domain
 maintainers     nomaintainer
-platforms       darwin
 description     A library providing routines for paper size management
 long_description \
                 The paper library and accompanying files are intended \
                 to provide a simple way for applications to take \
                 actions based on a system- or user-specified paper size.
 
-homepage        https://packages.qa.debian.org/libp/libpaper.html
-master_sites    debian:libp/${name}/
-
-distname        ${name}_${realversion}
-worksrcdir      ${name}-${realversion}
-
-checksums       rmd160  8ab9f62014375cc307d3b3a21b813af030032e11 \
-                sha256  26330e21e9a3124658d515fd850b0cde546ff42d89b2596a5264c5f1677f0547 \
-                size    44942
-
-post-patch {
-    set fd [open [file join ${worksrcpath}/lib paperspecs] a 0644]
-    # additional paper sizes
-    foreach {n x y} {jisb0 2920 4127
-                     jisb1 2064 2920
-                     jisb2 1460 2064
-                     jisb3 1032 1460
-                     jisb4  729 1032
-                     jisb5  516  729
-                     jisb6  363  516} {
-        puts ${fd} "${n} ${x} ${y}"
-    }
-    close ${fd}
-}
-
 use_autoreconf  yes
 
-configure.args  --mandir=${prefix}/share/man    \
-                --sysconfdir=${prefix}/etc
+configure.args  --enable-relocatable
+
+test.run        yes
+test.target     check
 
-livecheck.type      regex
-livecheck.version   ${realversion}
-livecheck.regex     {"stable">([^<]+)}
+post-destroot {
+    # remove `paper` and `paperconf`
```

Comment:
OK.  What's the best approach?  A sub-port?  A completely different one?

Note that it will take again about two weeks until I have again access to my Mac, so please be patient :slightly_smiling_face:

Reply:
Sorry, I neglected to reply to this earlier.

MacPorts was originally designed only for local source builds, so unlike other package managers it does not readily accommodate splitting a single build into multiple ports. I am not aware of many examples of this myself; maybe libxkbcommon-x11 is one example, although I would refrain from blaming upstream projects for refusing to accommodate MacPorts' limitations relative to other far more popular package managers.

But since the distfile is the same, a subport as you have done is likely better than a separate one.